[Dotnet] Starts a single server instance for integration tests.

batiati opened the pull request:
Changes the integration tests logic to share the same server instance for running all tests instead of a fresh instance per test.

It also removes the `[DoNotParallelize]` attribute.

Before
```
Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 2 m 45 s
```

After
```
Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 1 m 28 s
```

matklad commented:
🥳

matklad commented on src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:
```diff
@@ -142,94 +163,98 @@ public void ConstructorAndFinalizer()
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
```
Tbh, I’d drop these extra asserts from tests: this is something forced upon us by the testing framework, but we are using an idiomatic testing framework, and so can accept slight degradation of safety (in tests !!), so as not to swim upstream against C# here.

batiati replied:
It's due to the compiler null checks. I'd rather use those `Assert.IsNotNull(server)` than the null-forgiving syntax `server!` everywhere.

matklad replied:
Ahh, that makes sense. Seems like C# has an escape hatch here?

https://stackoverflow.com/a/73167373

Could we declare the relevant fields as non-null and init then with `null!`?

matklad commented on src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:
```diff
@@ -1540,38 +1563,38 @@ private void ConcurrentTasksDispose(bool isAsync)
 
 
-    private static void AssertAccounts(Account[] lookupAccounts)
+    private static void AssertAccounts(Account[] expected, Account[] actual)
```
Can we use this thing here? 

https://stackoverflow.com/a/11055686

hopefully it’ll print readable diff on mismatxhn

batiati replied:
We can't compare all fields; for example, the balances can change between creation and lookup.

batiati committed:
[Dotnet] Starts a single server instance for integration tests.

Changes the integration tests logic to share the same server instance for running all tests instead of a fresh instance per test.

It also removes the [DoNotParallelize] attribute.
